boards/shields/x_nucleo_iks4a1/doc/index.rst

```diff
@@ -0,0 +1,143 @@
+   development. It can be connected at the same time of external MEMS through
+   DIL24 adapter
+ - Available interface for external camera module applications coupled with
+   LSM6DSV16X through aux SPI (3/4 w)
+ - RoHS compliant
+ - WEEE compliant
+ - UKCA compliant
+
+Hardware Configuration
+**********************
+
+X-NUCLEO-IKS4A1 board can be configured in five different modes, which can be
+selected through J4 and J5 jumpers.
+
+.. _x-nucleo-iks4a1-mode-1:
+
+Mode 1: Standard Mode
+=====================
+
+In standard I2C mode, all devices are connected to an external main board via the
+same I2C bus.
+
+The board configuration is:
+
+ - J4: 1-2, 11-12 (STM_SDA = SENS_SDA, HUB_SDx = GND)
+ - J5: 1-2, 11-12 (STM_SCL = SENS_SCL, HUB_SCx = GND)
+
+.. _x-nucleo-iks4a1-mode-2:
+
+Mode 2: LSM6DSO16IS SensorHub Mode (SHUB2)
+==========================================
+
+In this sensor hub I2C mode, it is possible to power-up the 6-axes IMU
+(Inertial Measurement Unit) functionalities by collecting external data
+through a direct control of the on-board environmental sensors (temperature,
+pressure and magnetometer) and external sensor (DIL24) through the auxiliary
+I2Cz bus "SENS_I2C". LSM6DSV16X, LIS2DUXS12 and SHT40AD1B remains connected
+to the main bus "uC_I2C" coming from the external board.
+
+The board configuration is:
+
+ - J4: 7-8 (HUB2_SDx = SENS_SDA)
+ - J5: 7-8 (HUB2_SCx = SENS_SCL)
+
+.. _x-nucleo-iks4a1-mode-3:
+
+Mode 3: LSM6DSV16X SensorHub Mode (SHUB1)
+=========================================
+
+In this sensor hub, it is possible to power-up the 6-axes IMU (Inertial
+Measurement Unit) functionalities by collecting external data through
+a direct control of the on-board environmental sensors (temperature,
+pressure and magnetometer) and external sensor (DIL24) through the auxiliary
+I2C bus "SENS_I2C". LSM6DSO16IS, LIS2DUXS12 and SHT40AD1B remains connected
+to the main bus "uC_I2C" coming from the external board.
+
+The board configuration is:
+
+ - J4: 5-6 (HUB1_SDx = SENS_SDA)
+ - J5: 5-6 (HUB1_SDx = SENS_SDA)
+
+Mode 4: DIL24 SensorHub Mode
+============================
+
+In case a sensor with sensor hub embedded functionality is mounted to the
+board through DIL24 adapter, it is possible to exploit this functionality
+as for LSM6DSO16IS and the LSM6DSV16X. In this configuration, may be necessary
+to connect the DIL24 to the external board through SPI lines in order to
+avoid an address conflict on I2C bus with the LSM6DSO16IS and the LSM6DSV16X.
+This is done by changing the SBx configuration.
+
+The board configuration is:
+
+ - J4: 9-10 (DIL_SDx = SENS_SDA)
+ - J5: 9-10 (DIL_SDx = SENS_SDA)
+
+Mode 5: LSM6DSO16IS as Qvar controller
+======================================
+
+In this configuration, it is possible to use the equipped Qvar swipe electrode
+(by plugging it on JP6 and JP7 connectors) through the LSM6DSO16IS.
+
+The board configuration is:
+
+ - J4: 3-4 (HUB1_SDx = QVAR1)
+ - J5: 3-4 (HUB1_Scx = QVAR2)
+
```

Comment:
Some words about the specifics of the 3 overlays could be useful and how they interact with each other, if they do.

Reply:
OK, I'll try to describe more the three modes (and overlays) that this shield zephyr implementation is providing.

Reply:
ok, added some words... (not too many :D )